lib/reek/tree_walker.rb

```diff
@@ -105,17 +111,19 @@ def process_default(exp)
 
     def process_args(_) end
 
-    # :reek:TooManyStatements: { max_statements: 6 }
+    # :reek:TooManyStatements: { max_statements: 8 }
     # :reek:FeatureEnvy
     def process_send(exp)
+      method_name = exp.method_name
       if exp.visibility_modifier?
-        element.track_visibility(exp.method_name, exp.arg_names)
-      end
-      if exp.attribute_writer?
+        element.track_visibility(method_name, exp.arg_names)
+      elsif exp.attribute_writer?
         exp.args.each do |arg|
           next unless arg.type == :sym
           new_context(Context::AttributeContext, arg, exp)
         end
+      else
+        new_context(Context::SendContext, exp, method_name)
```

Comment:
>> Oh, I see now. Are you sure we need SendContext for that, and cannot just look for :send nodes using Reek::Ast::Node#find_nodes?

Technically we could, but I believe we shouldn't. First of all "defined_private_methods" works only on contexts, and it has to work on context because we don't now their visibility otherwise. Since the whole point of "method_calls" at least in the scope of this PR is to be able to do something like [here](https://github.com/troessner/reek/pull/724/files#diff-13f69a71ba371f43053cba8fecf2d1e0R64) this would break consistency and work on different levels of abstraction.

Conceptually I believe we should put more and more logic into our own abstractions. So our contexts in this case.
Or in other words: TreeWalker >>> TreeDresser for me. 
Our adorned AST is a paper thin abstraction that provides you with a bare minimum of features.
Additionally we already started to go down this road heavily by putting the visibility tracking in there.